twoliter/src/lock.rs

```diff
@@ -584,9 +626,48 @@ impl Lock {
             schema_version: project.schema_version(),
             sdk: LockedImage::new(&image_tool, vendor, sdk).await?,
             kit: locked,
+            vendor_override: Self::resolve_overrides(project).await?,
         })
     }
 
+    async fn resolve_overrides(project: &Project) -> Result<HashMap<String, Vendor>> {
```

Comment:
Do you mind adding docstrings to these new functions?

Unit tests on override resolution would also be helpful.